spec/models/manageiq/providers/base_manager/refresher_spec.rb

```diff
@@ -0,0 +1,84 @@
+RSpec.describe ManageIQ::Providers::BaseManager::Refresher do
+  context "#initialize" do
+    let(:ems1) { FactoryBot.create(:ext_management_system) }
+    let(:ems2) { FactoryBot.create(:ext_management_system) }
+    let(:vm1)  { FactoryBot.create(:vm, :ext_management_system => ems1) }
+    let(:vm2)  { FactoryBot.create(:vm, :ext_management_system => ems2) }
+
+    it "groups targets by ems" do
+      refresher = described_class.new([vm1, vm2])
+      expect(refresher.targets_by_ems_id.keys).to include(ems1.id, ems2.id)
+    end
+  end
+
+  context "#refresh" do
+    context "#preprocess_targets" do
```

Comment:
The nested contexts is weird to me...are we testing #refresh or are we testing #preprocess_targets ?